translate/storage/properties.py

```diff
@@ -1067,3 +1183,124 @@ class joomlafile(propfile):
     def __init__(self, *args, **kwargs):
         kwargs['personality'] = "joomla"
         super().__init__(*args, **kwargs)
+
+
+class XWikiPageProperties(xwikifile):
+    """
+    Represents an XWiki Page containing translation properties as described in
+    https://dev.xwiki.org/xwiki/bin/view/Community/XWiki%20Translations%20Formats/#HXWikiPageProperties
+    """
+
+    Name = "XWiki Page Properties"
+    Extensions = ['xml']
+    XML_HEADER = """<?xml version="1.1" encoding="UTF-8"?>
+
+    <!--
+     * See the NOTICE file distributed with this work for additional
+     * information regarding copyright ownership.
+     *
+     * This is free software; you can redistribute it and/or modify it
+     * under the terms of the GNU Lesser General Public License as
+     * published by the Free Software Foundation; either version 2.1 of
+     * the License, or (at your option) any later version.
+     *
+     * This software is distributed in the hope that it will be useful,
+     * but WITHOUT ANY WARRANTY; without even the implied warranty of
+     * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+     * Lesser General Public License for more details.
+     *
+     * You should have received a copy of the GNU Lesser General Public
+     * License along with this software; if not, write to the Free
+     * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+     * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+    -->
+
+    """
+
+    XWIKI_BASIC_XML = """<xwikidoc>
+    <translation>0</translation>
+    <language/>
+    <title/>
+    <content/>
+    </xwikidoc>
+    """
+
+    def __init__(self, *args, **kwargs):
+        kwargs['personality'] = "xwiki"
+        kwargs['encoding'] = "utf-8"
+        super(xwikifile, self).__init__(*args, **kwargs)
+        self.root = None
+
+    def parse(self, propsrc):
+        if propsrc != b"\n":
+            self.root = ElementTree.XML(propsrc)
+            content = ""\
+                .join(self.root.find("content").itertext())
+            content = unescape(content).encode(self.encoding)
+            super(XWikiPageProperties, self).parse(content)
+
+    def set_xwiki_xml_attributes(self, newroot):
+        for e in newroot.findall("object"):
+            newroot.remove(e)
+        for e in newroot.findall("attachment"):
+            newroot.remove(e)
```

Comment:
These removals are not covered by tests, are they needed?

Reply:
They are. Adding tests to cover it.

Reply:
It's now covered.